CMakeLists.txt

```diff
@@ -372,12 +372,13 @@ target_link_options(sanitize_interface INTERFACE ${SANITIZER_LDFLAGS})
 
 if(BUILD_FUZZ_BINARY)
   include(CheckSourceCompilesAndLinks)
-  check_cxx_source_compiles_with_flags("${SANITIZER_LDFLAGS}" "
+  check_cxx_source_compiles_with_flags("
       #include <cstdint>
       #include <cstddef>
       extern \"C\" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) { return 0; }
       // No main() function.
     " FUZZ_BINARY_LINKS_WITHOUT_MAIN_FUNCTION
+    CXXFLAGS ${SANITIZER_LDFLAGS}
```

Comment:
Could you add an LDFLAGS param to the function as `LINK_OPTIONS` and use that here instead?

Reply:
`LDFLAGS` might make more sense because, for example, in the OSS-Fuzz environment, the `SANITIZER_LDFLAGS` variable can be set to `-fsanitize=fuzzer` or `/usr/lib/libFuzzingEngine.a`.

Reply:
The `LDFLAGS` option has been [added](https://github.com/bitcoin/bitcoin/pull/31662#issuecomment-2593891189).

Reply:
> OSS-Fuzz

Wait, they're using a completely undocumented var. That shouldn't be any reason for us to change our behavior here. That's waaaay to brittle for us to have to worry about, no?

Why are they not using `CMAKE_EXE_LINKER_FLAGS` or `LDFLAGS` ?

Reply:
> Why are they not using `CMAKE_EXE_LINKER_FLAGS` or `LDFLAGS` ?

Both variants will work though.

cc the OSS-Fuzz crowd: @maflcko @dergoegge @fanquake @achow101

Reply:
Is it not an issue to link the fuzzer lib with our non-fuzz binaries? If not, yeah, let's just do that :)

Reply:
> Is it not an issue to link the fuzzer lib with our non-fuzz binaries?

I don't think so: https://github.com/bitcoin/bitcoin/blob/2d07384243c9552aa8e95c80d7a279e2d224a753/CMakeLists.txt#L211-L231

Reply:
Oh, right! Still.. I could imagine it breaking CMake checks. But sure, if it works, that'd be better than hacking around it here.